java/src/jmri/util/usb/UsbBrowserPanel.java

```diff
@@ -309,68 +424,84 @@ public Object getValueAt(int rowIndex, int columnIndex) {
                             break;
                     }
                     break;
+                }
                 case -1:
-                case 1:
+                case 1: {
                     try {
+                        UsbDevice usbDevice = node.getUsbDevice();
                         switch (rowIndex) {
                             case 0:
-                                return node.getUsbDevice().getManufacturerString();
+                                result = "?";
+                                return usbDevice.getManufacturerString();
                             case 1:
-                                return node.getUsbDevice().getProductString();
+                                result = "?";
+                                return usbDevice.getProductString();
                             case 2:
-                                return node.getUsbDevice().getSerialNumberString();
+                                result = "?";
+                                return usbDevice.getSerialNumberString();
                             case 3:
-                                return String.format("0x%04X", node.getUsbDevice().getUsbDeviceDescriptor().idVendor());
+                                result = "0x????";
+                                return String.format("0x%04X", usbDevice.getUsbDeviceDescriptor().idVendor());
                             case 4:
-                                return String.format("0x%04X", node.getUsbDevice().getUsbDeviceDescriptor().idProduct());
+                                result = "0x????";
+                                return String.format("0x%04X", usbDevice.getUsbDeviceDescriptor().idProduct());
                             case 5:
-                                return USBUtil.getLocationID(node.getUsbDevice());
+                                result = "?";
+                                return String.format("%d", usbDevice.getUsbDeviceDescriptor().bDeviceClass());
+                            case 6:
+                                result = "?";
+                                return String.format("%d", usbDevice.getUsbDeviceDescriptor().bDeviceSubClass());
+                            case 7:
+                                result = "?";
+                                return String.format("%d", usbDevice.getUsbDeviceDescriptor().bDeviceProtocol());
+                            case 8:
+                                log.info("* usbDevice: " + usbDevice);
+                                log.info("* usbDevice.getUsbDeviceDescriptor():\n" + usbDevice.getUsbDeviceDescriptor());
+                                result = "##.##";
+                                short version = usbDevice.getUsbDeviceDescriptor().bcdDevice();
+                                byte hiVersion = (byte) (version >> 8);
+                                byte loVersion = (byte) version;
+                                return String.format("%X.%02X", hiVersion, loVersion);
```

Comment:
Per the USB specification, this is a "binary coded decimal" (a four digit number). Why are you splitting it like this?

Reply:
Those are major.minor device release digits. If you log a UsbDevice.getUsbDeviceDescriptor() you'll get this:
  bLength                 18
  bDescriptorType          1
  bcdUSB                2.00
  bDeviceClass             9 Hub
  bDeviceSubClass          0
  bDeviceProtocol          1
  bMaxPacketSize0         64
  idVendor            0x05ac
  idProduct           0x1006
  bcdDevice            96.15
  iManufacturer            1
  iProduct                 2
  iSerial                  3
  bNumConfigurations       1

Note the bcdDevice is 96.15. That's the device (major.minor) release number in binary coded decimal.